src/containers/Signup/Signup.js

```diff
@@ -1,40 +1,55 @@
 import React from 'react';
 
 class Signup extends React.Component {
-    constructor (props) {
-        super(props)
-
-        this.state = {
-            inputs: {
-                username: "", 
-                email: "",
-                password: ""
-            }
-        }
+    state = {
+        inputs: {
+            username: "", 
+            email: "",
+            password: ""
+        },
+        fillFormInputError: ""
     }
 
     handleInputChange = e => {
         const { inputs } = {...this.state};
         inputs[e.target.name]= e.target.value.trim();
 
-        this.setState({ inputs })
+        this.setState({ inputs });
     }
 
-    render () {
+    validateFormInputs = () => {
         const { inputs } = this.state;
-        const inputArray = Object.entries(inputs);
+        
+        return inputs.email.length > 0 && inputs.password.length > 0;
+    }
+
+    handleSignupButtonClick = e => {
```

Comment:
You don't have to do this in JavaScript: 

* General guide: https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation
* Example docs, for number inputs: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#Validation